backend/tests/Unit/Http/Controller/AiRequestControllerTest.php

```
@@ -0,0 +1,190 @@
+            self::fail('Missing request must fail.');
+        } catch (NotFoundHttpException $error) {
+            self::assertSame('Request not found', $error->getMessage());
+        }
+    }
+
+    public function testDelegatesToSpecializedAiLifecycle(): void
+    {
+        $controller = $this->controller();
+        Yii::$container->set(AiRequestLifecycle::class, new class implements AiRequestLifecycle {
+            public function execute(
+                int $actorId,
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$actorId'. (undefined)

(UnusedFormalParameter)